sources/mongodb/helpers.py

```diff
@@ -198,13 +202,50 @@ def load_documents(self, limit: Optional[int] = None) -> Iterator[TDataItem]:
             yield document
 
 
+class CollectionArrowLoader(CollectionLoader):
+    """
+    Mongo DB collection loader, which uses
+    Apache Arrow for data processing.
+    """
+
+    def load_documents(self, limit: Optional[int] = None) -> Iterator[Any]:
+        """
+        Load documents from the collection in Apache Arrow format.
+
+        Args:
+            limit (Optional[int]): The number of documents to load.
+
+        Yields:
+            Iterator[Any]: An iterator of the loaded documents.
+        """
+        context = PyMongoArrowContext.from_schema(
+            None, codec_options=self.collection.codec_options
+        )
+
+        cursor = self.collection.find_raw_batches(
+            self._filter_op, batch_size=self.chunk_size
+        )
+        if self._sort_op:
+            cursor = cursor.sort(self._sort_op)  # type: ignore
+
+        cursor = self._limit(cursor, limit)  # type: ignore
+
+        for batch in cursor:
+            process_bson_stream(batch, context)
+
+            table = context.finish().to_pylist()
+
+            yield map_nested_in_place(convert_mongo_objs, table)
```

Comment:
So far, I tried to use schema for the `find_arrow_all()`, schema in the context and a single column type cast - none work. Whether it fails with an error or turns all the values into `null` (the column type changes though)